admin/python/lsst/qserv/admin/nodePool.py

```diff
@@ -0,0 +1,160 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+Module defining nodePool class and related methods.
+
+NodePool is a class which allow to launch parallel ssh commands to
+a set of NodeAdmin instances. Each NodeAdmin instance can be a Qserv
+master or worker node.
+
+It has been used at CC-IN2P3 in spring 2015 to install and manage multiple
+Qserv development versions for a 50 nodes cluster.
+
+@author  Fabrice Jammes, IN2P3
+"""
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import logging
+import os
+from subprocess import Popen, list2cmdline
+import time
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+_LOG = logging.getLogger(__name__)
+
+def cpu_count():
+    ''' Returns the number of CPUs in the system
+    '''
+    try:
+        return os.sysconf('SC_NPROCESSORS_ONLN')
+    except (ValueError, OSError, AttributeError):
+        return 1
+
+class NodePool(object):
+    """
+    Class providing tools to send ssh command to a set of machines.
+    Use system ssh.
+    """
+
+    def __init__(self, nodeAdmins=None, max_task=None):
+        """
+        Container of nodes which will be managed in parallel using ssh
+
+        @param nodeAdmins:  a list of NodeAdmin instances, each node has to be
+                            contactable via SSH
+        @param max_task:    number of parallel SSH command which will be run
+                            concurrently, equal to the number of cpu of the local
+                            machine by default
+        """
+
+        self.nodes = nodeAdmins
+
+        if max_task:
+            self._max_task = max_task
+        else:
+            self._max_task = cpu_count()
+
+    def execParallel(self, command, stdin=os.devnull):
+        """
+        Execute one SSH command in parallel on a set of nodes;
+        fork one process for each node.
+        Log of each command are stored in $CWD/parallel-cmd-log
+
+        @param command: command to run
+        @param stdin: file redirect to SSH client process standard input
+
+        @return the number of nodes in error
+        """
+        nodes_remaining = list(self.nodes)
+
+        if not command:
+            return  # empty list
+
+        logdir = "parallel-cmd-log"
+        try:
+            os.makedirs(logdir)
+        except OSError:
+            if not os.path.isdir(logdir):
+                raise
+
+        def logfile(name):
+            f = "{0}-{1}-{2}.txt".format(node.host, process_id, name)
```

Comment:
Why do you pass name as argument but do not pass process_id or node (or node.host)?